components/scream/src/control/tests/sc_tests.cpp

```diff
@@ -139,18 +154,276 @@ TEST_CASE ("surface_coupling")
+      REQUIRE (v2d_exp_h(icol,0)==v2d_imp_h(icol,0));
+      REQUIRE (v2d_exp_h(icol,1)==v2d_imp_h(icol,1));
+      REQUIRE (v3d_exp_h(icol,0,nlevs-1)==v3d_imp_h(icol,0,nlevs-1));
+      REQUIRE (v3d_exp_h(icol,1,nlevs-1)==v3d_imp_h(icol,1,nlevs-1));
     }
   }
 
   // Clean up
   delete[] raw_data;
 }
+
+
+TEST_CASE ("recreate_mct_coupling")
+{
+  /*
+   * This test aims to recreate the import/export of fields inside the mct-coupling
+   * for CIME runs. Much of the code inside surface coupling is hard-coded for these runs.
+   */
+
+  // Some namespaces/aliases
+  using namespace scream;
+  using namespace ShortFieldTagsNames;
+  using namespace ekat::units;
+  using FL     = FieldLayout;
+  using FID    = FieldIdentifier;
+  using FR     = FieldRequest;
+  using GR     = GroupRequest;
+  using RPDF   = std::uniform_real_distribution<Real>;
+  using rngAlg = std::mt19937_64;
+
+  // Some constants
+  constexpr int ncols = 4;
+  constexpr int nlevs = 8;
+  constexpr int nruns = 10;
+
+  // Create a comm
+  ekat::Comm comm (MPI_COMM_WORLD);
+
+  // The random numbers generator
+  std::random_device rd;
+  const unsigned int catchRngSeed = Catch::rngSeed();
+  const unsigned int seed = catchRngSeed==0 ? rd() : catchRngSeed;
+  if (comm.am_i_root()) {
+    std::cout << "seed: " << seed << (catchRngSeed==0 ? " (catch rng seed was 0)\n" : "\n");
+  }
+  rngAlg engine(seed);
+  RPDF pdf(0.0,1.0);
+
+  // Create a grid
+  auto grid = create_point_grid("my_grid",ncols*comm.size(), nlevs, comm);
+  const auto grid_name = grid->name();
+
+  // Layouts matching those in AD
+  FL scalar2d_layout{ {COL          }, {ncols          } };
+  FL scalar3d_layout{ {COL, LEV     }, {ncols,    nlevs} };
+  FL vector3d_layout{ {COL, CMP, LEV}, {ncols, 2, nlevs} };
+
+  // Create import fields
+  FID surf_latent_flux_id ("surf_latent_flux", scalar2d_layout, W/(m*m), grid_name);
+  FID surf_sens_flux_id   ("surf_sens_flux",   scalar2d_layout, W/(m*m), grid_name);
+  FID surf_u_mom_flux_id  ("surf_u_mom_flux",  scalar2d_layout, W/(m*m), grid_name);
+  FID surf_v_mom_flux_id  ("surf_v_mom_flux",  scalar2d_layout, W/(m*m), grid_name);
+
+  // Create necessary fields for export. Tracers qc and qr are unnecessary, but
+  // are included to verify that subviewed fields (qv) are correctly handled
+  const auto nondim = Units::nondimensional();
+  FID T_mid_id           ("T_mid",           scalar3d_layout, K,      grid_name);
+  FID p_mid_id           ("p_mid",           scalar3d_layout, Pa,     grid_name);
+  FID z_mid_id           ("z_mid",           scalar3d_layout, m,      grid_name);
+  FID horiz_winds_id     ("horiz_winds",     vector3d_layout, m/s,    grid_name);
+  FID pseudo_density_id  ("pseudo_density",  scalar3d_layout, Pa,     grid_name);
+  FID qv_id              ("qv",              scalar3d_layout, nondim, grid_name);
+  FID qc_id              ("qc",              scalar3d_layout, nondim, grid_name);
```

Comment:
Is there a reason to register these? I don't see them used here.

Reply:
I just included in case there was some issue with more than one field in a group. I guess this is being tested though in the first test. I'll remove.